### DISCUSSION TITLE
Hoover Washing Machine HBWOS 69-TAME-80

### DISCUSSION THREAD

**Discussion by jonshorter** [ofalvai/home-assistant-candy#293] [Device support improvements]
Hi @ofalvai thanks for your work on this project - here's what I've worked out for the Hoover Washer HBWOS 69-TAME-80...

```
class WashProgramType(StatusCode):
    EXTRA_CARE = (1, "Extra Care")
    AIO_59 = (2, "All In One 59'")
    RAPID_143044 = (3, "Rapid Care 14'//30'//44'")
    ALLERGY_CARE = (4, "Allergy Care 60°C")
    FRESH_CARE = (5, "Fresh Care")
    SOFT_CARE = (6, "Soft Care")
    FITNESS_CARE = (7, "Fitness Care")
    RINSE_PR = (8, "Rinse")
    DRAIN_SPIN = (9, "Drain & Spin")
    MIXED = (10, "Mixed")
    ECO20 = (11, "Eco 20°C")
    WOOL_HAND = (12, "Wool & Handwash")
    ECO_4060 = (13, "Eco 40°C-60°C")
    COTTONS = (14, "Cottons")
    ONE_FI = (15, "One Fi Extra")
```

Here are the stain-level values (SLevel):
1:Lightly Stained
2:Mildly Stained
3:Heavily Stained

Here is the spin-speed format (SpinSp):
12 = 1200rpm (/100)
4 = 400rpm ... etc ...

Here is the remaining time format (RemTime):
60 = 60 minutes (not seconds)

Here is the temperature format (Temp):
40 = 40°C

Here's the output for this model (encrypted=1, the hex is plaintext)...

```
"statusLavatrice":{
                "WiFiStatus":"0",
                "Err":"255",
                "MachMd":"2",
                "Pr":"10",
                "PrPh":"2",
                "SLevel":"3",
                "Temp":"40",
                "SpinSp":"12",
                "Opt1":"0",
                "Opt2":"0",
                "Opt3":"0",
                "Opt4":"0",
                "Opt5":"0",
                "Opt6":"0",
                "Opt7":"0",
                "Opt8":"0",
                "Steam":"0",
                "DryT":"0",
                "DelVal":"255",
                "RemTime":"76",
                "RecipeId":"0",
                "CheckUpState":"0"
```

Jonathan

**Comment by ofalvai** [comment]
Hey @jonshorter, the programs (`Pr` in the payload) are not parsed at the moment because they differ between models. The program is exposed as an int though, so you could create a templated entity with your specific mapping.

Unfortunately, some other fields are also model-dependent, looks like your is one of those that report the remaining time as minutes and not as seconds. See these related issues:
- https://github.com/ofalvai/home-assistant-candy/issues/252
- https://github.com/ofalvai/home-assistant-candy/issues/193
